<launch>
    <arg name="run_rviz"              default="true"/>
    <arg name="run_slam_live"         default="true"/>
    <arg name="run_reconfigure"       default="true"/>

    <arg name="camera" default="camera"/>
    
    <arg name="debug" default="false" />
    <arg name="launch_prefix" value="xterm -e gdb --args" if="$(arg debug)" />
    <arg name="launch_prefix" value="" unless="$(arg debug)" />
    <node launch-prefix="$(arg launch_prefix)" pkg="dvo_ros" type="camera_tracker" name="camera_tracker" output="screen" required="true">
        <!-- sensor re-mappings for UNC Charlotte bag data (default) -->
        <remap from="rgb_topic" to="/camera/rgb/input_image"/>
        <remap from="depth_topic" to="/camera/depth_registered/input_image"/>
        <remap from="camera_info_topic" to="/camera/rgb/camera_info"/>        
    </node>

    <!-- RVIZ SECTION -->
    <!--<group unless="$(arg backend_only)">-->
        <arg name="rviz_config_file" default="dvo_ros.rviz"/>
        <node pkg="rviz" type="rviz" name="rviz" output="screen" 
              args="-d $(find dvo_ros)/rviz/$(arg rviz_config_file)" if="$(arg run_rviz)"/>
    <!--</group>-->

    <!-- DYNAMIC RECONFIGURE SECTION -->
    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" output="screen" if="$(arg run_reconfigure)"/> 

    <group ns="$(arg camera)">
        <param name="driver/data_skip" value="0" />
        <remap from="depth_registered/image_raw" to="depth_registered/input_image"/>    
        <remap from="rgb/image_raw" to="rgb/input_image"/> 
        <!--<remap from="depth_registered/image_raw" to="depth_registered/image_rect_raw"/>    
        <remap from="rgb/image_raw" to="rgb/image_rect"/>
        <remap from="rgb/camera_info" to="rgb/camera_info"/>-->
        <!--<remap from="depth_registered/camera_info" to="depth_registered/camera_info"/>-->

        
        <include if="$(arg run_slam_live)"
                 file="$(find rgbd_launch)/launch/includes/manager.launch.xml">
            <arg name="name" value="$(arg camera)_nodelet_manager" />
            <arg name="debug" value="false" />
            <arg name="num_worker_threads"  value="4" />
        </include>

        <!-- Load astra driver -->
        <include if="$(arg run_slam_live)"
                 file="$(find astra_launch)/launch/includes/device.launch.xml">            
            <arg name="manager"                         value="$(arg camera)_nodelet_manager" />
            <arg name="device_id"                       value="#1" />
            <arg name="bootorder"                       value="0" />
            <arg name="devnums"                         value="1" />
            <arg name="rgb_frame_id"                    value="$(arg camera)_rgb_optical_frame" />
            <arg name="depth_frame_id"                  value="$(arg camera)_depth_optical_frame" />
            <arg name="rgb_camera_info_url"             value="" />
            <arg name="depth_camera_info_url"           value="" />
            <arg name="rgb"                             value="rgb" />
            <arg name="ir"                              value="ir" />
            <arg name="depth"                           value="depth" />
            <arg name="respawn"                         value="false" />
            <arg name="depth_registration"              value="true" />
            <arg name="color_depth_synchronization"     value="false" />
            <arg name="auto_exposure"                   value="true" />
            <arg name="auto_white_balance"              value="true" />
        </include>
        
    </group>
</launch>
